### PR TITLE
Call to close() replaced by warning in Realm finalizer.

### PR DESCRIPTION
**Pull request by cmelchior**
Calling close() in finalizer led to people getting warnings even though they correctly closed their Realm instance. Now a warning is emitted instead, much like Android cursors. Reported here: https://github.com/realm/realm-java/issues/763

No unit test accompanies this as unit testing finalisers is as far as I know impossible without incurring a huge and unacceptable time penalty.

### CODE REVIEW

**Comment by cmelchior** [comment]
@emanuelez @kneth @bmunkholm

File: realm/src/main/java/io/realm/Realm.java [hunk 1]
```diff
@@ -166,7 +166,9 @@ private Realm(String absolutePath, byte[] key, boolean autoRefresh) {
 
     @Override
     protected void finalize() throws Throwable {
-        this.close();
+        if (sharedGroup != null) {
+            RealmLog.w("Realm " + path + " is being finalized without being closed. This can lead to running out of memory.");
```
Comment (emanuelez) [inline realm/src/main/java/io/realm/Realm.java]: I would maybe specify *native* memory.

Comment (bmunkholm) [inline realm/src/main/java/io/realm/Realm.java]: Maybe add "Remember to call close() on all realm instances."

**Comment by bmunkholm** [comment]
miss update of changelog

**Pull request merged**